platform/viewer/src/connectedComponents/ToolbarRow.js

```diff
@@ -279,21 +267,38 @@ function _getVisibleToolbarButtons() {
   return toolbarButtonDefinitions;
 }
 
-function _handleBuiltIn({ behavior } = {}) {
+function _handleBuiltIn({ id, options: { behavior } } = {}) {
+  /* TODO: Keep cine button active until its unselected. */
+  const { dialog, modal, t } = this.props;
+  const { dialogId } = this.state;
+
   if (behavior === 'CINE') {
-    this.setState({
-      isCineDialogOpen: !this.state.isCineDialogOpen,
-    });
+    if (dialogId) {
+      dialog.dismiss({ id: dialogId });
+      this.setState({ dialogId: null });
+    } else {
+      const spacing = 20;
+      const { x, y } = document
+        .querySelector(`.ViewerMain`)
+        .getBoundingClientRect();
+      const newDialogId = dialog.create({
+        content: ConnectedCineDialog,
+        defaultPosition: {
+          x: x + spacing || 0,
+          y: y + spacing || 0,
+        },
+      });
+      this.setState({ dialogId: newDialogId });
+    }
   }
 
   if (behavior === 'DOWNLOAD_SCREEN_SHOT') {
-    this.props.modal.show(ConnectedViewportDownloadForm, {
-      title: this.props.t('Download High Quality Image'),
-      customClassName: 'ViewportDownloadForm',
+    modal.show(ConnectedViewportDownloadForm, {
```

Comment:
Similarly, I think we can move this to the `@ohif/extension-cornerstone` and ditch the built in here (outside scope of this PR). This one's a bit closer/easier.

We use `state.viewports` to grab `viewportSpecificData` and `activeViewportIndex`
We use those to grab `dom` off the `viewportSpecificData`'s `activeViewport`

DOM is just a reference to the active viewport's `enabledElement`
`dom` is actually set/added by the `@ohif/extension-cornerstone`

The only bit of data we need exposed to refactor the `ConnectedViewportDownloadForm` _should_ be the `activeViewportIndex` which we already have exposed for extensions (albeit, that method could be cleaned up).

Reply:
Backlogged here: https://github.com/OHIF/Viewers/issues/1206